clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -1715,8 +1717,10 @@ template <typename RecordType, typename ReaderType>
 typename scheduler_tmpl_t<RecordType, ReaderType>::stream_status_t
 scheduler_tmpl_t<RecordType, ReaderType>::next_record(output_ordinal_t output,
                                                       RecordType &record,
-                                                      input_info_t *&input)
+                                                      input_info_t *&input,
+                                                      uint64_t cur_time)
 {
+    outputs_[output].cur_time = cur_time;
```

Comment:
IMHO that will cost more in synchronization overhead than it will be worth: there is already a check versus the current input's start time.  I added a comment about deliberately not doing a global check here and only having that local input check.